AnnService/src/Core/Common/DistanceUtils.cpp

```diff
@@ -259,7 +278,42 @@ float DistanceUtils::ComputeL2Distance_AVX(const std::int8_t* pX, const std::int
     return diff;
 }
 
+float DistanceUtils::ComputeL2Distance(const std::uint8_t* pX, const std::uint8_t* pY, DimensionType length) 
+{
+    if (DistanceUtils::Quantizer != nullptr) {
+        return DistanceUtils::Quantizer->L2Distance(pX, pY);
+    }
+    return DistanceUtils::ComputeL2Distance_NonQuantized(pX, pY, length);
```

Comment:
Could you do the distance function selection in the DistanceCalcSelector instead of check "if" each time?